velox/expression/ExprCompiler.cpp

```diff
@@ -329,6 +329,9 @@ ExprPtr tryFoldIfConstant(const ExprPtr& expr, Scope* scope) {
     // instance, if other arguments are all null in a function with default null
     // behavior), the query won't fail.
     catch (const std::exception&) {
+      // Clear left-behind values. Simplified path will error out if
```

Comment:
I'm not sure I understand this. Also, looks like this code path is not covered by tests. I think we should catch only VeloxUserException and add test cases to ExprTes, ExprCompilerTest, and EvalSimplifiedTest.

CC: @bikramSingh91 @kagamiori